homeassistant/components/fan/wemo.py

```diff
@@ -0,0 +1,223 @@
+
+        # look up model name once as it incurs network traffic
+        self._model_name = self.wemo.model_name
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register update callback."""
+        wemo = get_component('wemo')
+        # The register method uses a threading condition, so call via executor,
+        # and yield from to wait until the task is done.
+        yield from self.hass.async_add_job(
+            wemo.SUBSCRIPTION_REGISTRY.register, self.wemo)
+        # The on method just appends to a defaultdict list.
+        wemo.SUBSCRIPTION_REGISTRY.on(self.wemo, None, self._update_callback)
+
+    def _update_callback(self, _device, _type, _params):
+        """Update the state by the Wemo device."""
+        _LOGGER.debug("Subscription update for  %s", _device)
+        updated = self.wemo.subscription_update(_type, _params)
+        self._update(force_update=(not updated))
+        self.schedule_update_ha_state()
+
+    def _update(self, force_update=True):
+        """Update the device state."""
+        try:
+            self._state = self.wemo.get_state(force_update)
+
+            self._fan_mode = self.wemo.fan_mode_string
+            self._target_humidity = self.wemo.desired_humidity_percent
+            self._current_humidity = self.wemo.current_humidity_percent
+            self._water_level = self.wemo.water_level_string
+            self._filter_life = self.wemo.filter_life_percent
+            self._filter_expired = self.wemo.filter_expired
+
+            if self.wemo.fan_mode != WEMO_FAN_OFF:
+                self._last_fan_on_mode = self.wemo.fan_mode
+        except AttributeError as err:
+            _LOGGER.warning("Could not update status for %s (%s)",
+                            self.name, err)
+
+    @property
+    def should_poll(self):
+        """No polling needed with subscriptions."""
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the ID of this WeMo humidifier."""
+        return "{}.{}".format(self.__class__, self.wemo.serialnumber)
+
+    @property
+    def name(self):
+        """Return the name of the switch if any."""
+        return self.wemo.name
+
+    @property
+    def is_on(self):
+        """Return true if switch is on. Standby is on."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon of device based on its type."""
+        return 'mdi:water-percent'
+
+    def turn_on(self: ToggleEntity, speed: str=None, **kwargs) -> None:
+        """Turn the switch on."""
+        if speed is None:
+            self.wemo.set_state(self._last_fan_on_mode)
+        else:
+            self.set_speed(speed)
+
+    def turn_off(self: ToggleEntity, **kwargs) -> None:
+        """Turn the switch off."""
+        self.wemo.set_state(WEMO_FAN_OFF)
+
+    def set_speed(self: ToggleEntity, speed: str) -> None:
+        """Set the fan_mode of the Humidifier."""
+        if speed == SPEED_OFF:
```

Comment:
You could use a dict here too, which is the inverse of the other state dict, that maps the home assistant speed to the wemo speed.